extensions/amp-ad-network-doubleclick-impl/0.1/safeframe-host.js

```diff
@@ -570,37 +570,37 @@ export class SafeframeHostApi {
    */
   resizeAmpAdAndSafeframe(height, width, messageType, optIsCollapse) {
     this.baseInstance_.attemptChangeSize(height, width).then(() => {
-      const success = !!this.baseInstance_.element.style.height.match(height)
-              && !!this.baseInstance_.element.style.width.match(width);
-      // If the amp-ad element was successfully resized, always update
-      // the size of the safeframe as well. If the amp-ad element could not
-      // be resized, but this is a collapse request, then only collapse
-      // the safeframe.
-      if (success || optIsCollapse) {
-        this.resizeIframe(height, width);
-        this.isCollapsed_ = !!optIsCollapse;
-        this.baseInstance_.element.getResources().resources_.forEach(
-            resource => {
-              if (resource.element == this.baseInstance_.element) {
-                // Need to force a measure event, as measure won't happen immediately
-                // if the element was above the viewport when resize occured, and
-                // without a measure, we'll send the wrong size for the creative
-                // on the geometry update message.
-                resource.measure();
-              }
-            });
-      } else {
-        // attemptChangeSize automatically registers a pendingChangeSize if
-        // the initial attempt failed. We do not want to do that, so clear it.
-        this.baseInstance_.element.getResources().resources_.forEach(
-            resource => {
-              if (resource.element == this.baseInstance_.element) {
-                resource.pendingChangeSize_ = undefined;
-              }
-            });
+      try {
+        const success =
+              (this.baseInstance_.element.style.height == height + 'px')
+              && (this.baseInstance_.element.style.width == width + 'px');
+        // If the amp-ad element was successfully resized, always update
+        // the size of the safeframe as well. If the amp-ad element could not
+        // be resized, but this is a collapse request, then only collapse
+        // the safeframe. Collapse will always resize the safeframe.
+        if (success || optIsCollapse) {
+          this.resizeIframe(height, width);
+          this.isCollapsed_ = !!optIsCollapse;
+          // We need to force a measure right now, as without remeasuring, we
+          // will have the wrong size information for the amp-ad and the
+          // safeframe when we send the response message into the safeframe.
+          // This would be an issue specifically for when expand is successful
+          // while amp-ad is out of the viewport, for example.
+          this.baseInstance_.getResource().measure();
```

Comment:
I don't understand this. If it's been resized, it should have new measurements.

Further, this will really hurt in layers system. Given most ads are in the root scroller, this will cause a remeasure of **every** amp-element on the page.

Reply:
When the safeframe is scrolled out of page, if we don't explicitly remeasure, then we get the wrong measurements when we send the update into the safeframe, i.e. we still get the old measurements of the frame.